sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -110,7 +110,7 @@ case class DataSource(
     }
   }
 
-  private def providingInstance() = providingClass.getConstructor().newInstance()
+  def providingInstance(): Any = providingClass.getConstructor().newInstance()
```

Comment:
Would it be possible to still mark it as private[sql] or private[spark]?

Reply:
Done!